### ISSUE TITLE
Core_kernel-based project fails to link

### ISSUE THREAD

**Issue by anton-trunov** [ocamllabs/ocaml-multicore#460]
Hi, I'm not sure whether this is the right place to report this issue, I apologize if it's not and will happily move it elsewhere.

I have set up my `multicore-ocaml` environment based (I rely on the `Core_kernel` library) following the instructions [here](https://github.com/ocaml-multicore/multicore-opam/issues/44#issuecomment-747667076). Now I'm trying to run a hello-world example using the following sample project:

```shell
$ tree hello
hello
├── dune
└── hello.ml

$ cat hello/hello.ml
open Core_kernel
let () = print_endline "Hello, World!"

$ cat hello/dune
(executable
 (name hello)
 (libraries core_kernel))
```

but I run into a linking error:

```shell
$ dune exec -- hello/hello.exe
    ocamlopt hello/hello.exe (exit 2)
(cd _build/default && /Users/anton/.opam/4.10.0+multicore+no-effect/bin/ocamlopt.opt -w @1..3@5..28@30..39@43@46..47@49..57@61..62-40 -strict-sequence -strict-formats -short-paths -keep-locs -g -o hello/hello.exe /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/base_internalhash_types/base_internalhash_types.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/base_internalhash_types /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/caml/caml.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/sexplib0/sexplib0.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/shadow_stdlib/shadow_stdlib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/base.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_sexp_conv/runtime-lib/ppx_sexp_conv_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_compare/runtime-lib/ppx_compare_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_enumerate/runtime-lib/ppx_enumerate_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_hash/runtime-lib/ppx_hash_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_here/runtime-lib/ppx_here_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_assert/runtime-lib/ppx_assert_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_bench/runtime-lib/ppx_bench_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base/md5/md5_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ocaml/unix.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ocaml /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ocaml/bigarray.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ocaml /Users/anton/.opam/4.10.0+multicore+no-effect/lib/fieldslib/fieldslib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/variantslib/variantslib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/bin_prot/shape/bin_shape_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/bin_prot/bin_prot.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/bin_prot /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_inline_test/config/inline_test_config.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/jane-street-headers/jane_street_headers.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/time_now/time_now.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/time_now /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_inline_test/runtime-lib/ppx_inline_test_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/stdio/stdio.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_module_timer/runtime/ppx_module_timer_runtime.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/typerep/typerep_lib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_expect/common/expect_test_common.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_expect/config_types/expect_test_config_types.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_expect/collector/expect_test_collector.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_expect/collector /Users/anton/.opam/4.10.0+multicore+no-effect/lib/ppx_expect/config/expect_test_config.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/splittable_random/splittable_random.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base_quickcheck/base_quickcheck.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base_bigstring/base_bigstring.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/base_bigstring /Users/anton/.opam/4.10.0+multicore+no-effect/lib/core_kernel/base_for_tests/base_for_tests.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/parsexp/parsexp.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/sexplib/sexplib.cmxa /Users/anton/.opam/4.10.0+multicore+no-effect/lib/core_kernel/core_kernel.cmxa -I /Users/anton/.opam/4.10.0+multicore+no-effect/lib/core_kernel hello/.hello.eobjs/native/dune__exe__Hello.cmx)
Undefined symbols for architecture x86_64:
  "_caml_adjust_gc_speed", referenced from:
      _bigstring_alloc in libbase_bigstring_stubs.a(base_bigstring_stubs.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
File "caml_startup", line 1:
Error: Error during linking
```

My setup:
- macOS 11.0.1 Big Sur,
- OCaml, version 4.10.0+multicore+no-effect.

I also have an opam switch based on the plain OCaml v4.10.0 and it can compile, link, and run the sample project as expected.

**Comment by ctk21** [comment]
Thanks for the report it is useful. 

We believe this to be fixed post the merge of #589; please do file issues if you find new problems with `Core_kernel` and `4.12+domains`

**Comment by anton-trunov** [comment]
@ctk21 Thanks a lot!
